tests/test_permissions.py

```diff
@@ -36,20 +36,19 @@ def test_create_permission_fail_if_group_already_exists(mocker):
             status=200,
         )
 
-        artifactory_permission = ArtfictoryPermission(AuthModel(url=URL, auth=AUTH))
+        artifactory_permission = ArtifactoryPermission(AuthModel(url=URL, auth=AUTH))
         mocker.spy(artifactory_permission, "get")
         with pytest.raises(PermissionAlreadyExistsException):
-            artifactory_permission.create(PERMISSION)
-
-        artifactory_permission.get.assert_called_once_with(PERMISSION.name)
+            permission = artifactory_permission.create(PERMISSION)
+            artifactory_permission.get.assert_called_once_with(PERMISSION.name)
+            assert permission == PERMISSION.dict()

```

Comment:
If the line 42 actually raises an Exception, I don't think the other tests run, so they should be moved out of the `with` block.
Also, `create_permission` raises if the permission already exists, so it can't return a `permission`, so you can't check if it's the same as what you wanted to create (which means the last `assert` is not useful and should even break once out of the `with` block).